tsconfig.json

```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    // Tells TypeScript to read JS files, as
+    // normally they are ignored as source files
+    "allowJs": true,
+    // Generate d.ts files
+    "declaration": true,
+    // This compiler run should
+    // only output d.ts files
+    "emitDeclarationOnly": true,
+    "jsx": "react",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    // Types should go into this directory.
+    // Removing this would place the .d.ts files
+    // next to the .js files
+    "outDir": "dist"
+  },
+  "files": ["./src/typings/declarations.d.ts"],
+  "include": ["src/**/*"],
+  "exclude": ["node_modules", "src/legacy"]
+}
```

Comment:
These are all experimentations and possibly all or none of these options can be used.

Reply:
I think most of these settings look reasonable to me. I think it'd be a good idea, however, for us to go through the available options and decide which ones we want to enable/disable. Since I haven't really worked in a TypeScript project before, I'm not really familiar with all the options, but I do know about the [strict null checks option](https://www.typescriptlang.org/tsconfig#strictNullChecks) and I absolutely think we should enable that one.

@jessypeck, do you have any thoughts on a good set of options to enable? My preference is to err on the side of being overly strict, since it's easier to relax them over time than to increase strictness.